src/Driver.cpp

```diff
@@ -154,78 +156,94 @@ int Driver::getFileDescriptor() const
 }
 bool Driver::isValid() const { return m_stream; }
 
-void Driver::openURI(std::string const& uri)
-{
+void Driver::openURI(std::string const& uri) {
     // Modes:
     //   0 for serial
     //   1 for TCP
     //   2 for UDP
     //   3 for UDP server
     //   4 for file (either Unix sockets or named FIFOs)
     int mode_idx = -1;
-    char const* modes[6] = { "serial://", "tcp://", "udp://", "udpserver://", "file://", "test://" };
-    for (int i = 0; i < 6; ++i)
-    {
-        if (uri.compare(0, strlen(modes[i]), modes[i]) == 0)
-        {
+    char const* modes[6] = { "serial://", "tcp://", "udp://",
+                             "udpserver://", "file://", "test://" };
+    for (int i = 0; i < 6; ++i) {
+        if (uri.compare(0, strlen(modes[i]), modes[i]) == 0) {
             mode_idx = i;
             break;
         }
     }
 
-    if (mode_idx == -1)
+    if (mode_idx == -1) {
         throw std::runtime_error("unknown URI " + uri);
-
+    }
     string device = uri.substr(strlen(modes[mode_idx]));
 
     // Find a :[additional_info] marker
     string::size_type marker = device.find_last_of(":");
+
     int additional_info = 0;
-    if (marker != string::npos)
-    {
-        additional_info = boost::lexical_cast<int>(device.substr(marker + 1));
+    SerialConfiguration serial_config;
+    bool has_serial_config = false;
+
+    if (marker != string::npos) {
+        if (mode_idx == 0) {
+            string info = device.substr(marker + 1);
+            device = device.substr(0, marker);
+
+            string::size_type baud_marker = device.find_last_of(":");
+            if (baud_marker != string::npos) {
+                additional_info = boost::lexical_cast<int>(
+                    device.substr(baud_marker + 1)
+                );
+
+                serial_config = parseSerialConfiguration(info);
+                has_serial_config = true;
+                device = device.substr(0, baud_marker);
+            } else {
+                additional_info = boost::lexical_cast<int>(info);
+            }
+        } else {
+            additional_info = boost::lexical_cast<int>(device.substr(marker + 1));
+        }
         device = device.substr(0, marker);
     }
 
-    if (mode_idx == 0)
-    { // serial://DEVICE:baudrate
-        if (marker == string::npos)
+    if (mode_idx == 0) { // serial://DEVICE:baudrate
+        if (marker == string::npos) {
             throw std::runtime_error("missing baudrate specification in serial:// URI");
-        openSerial(device, additional_info);
+        }
+        openSerial(device, additional_info, has_serial_config ? &serial_config : NULL);
```

Comment:
There is always a serial config ... the default, which is 8N1. Make it the default for SerialConfig and avoid having to pass a pointer.

Reply:
Yeah, I avoided that because I think it might break backwards compatibility. What do you think?

Reply:
'raw' is 8N. So, that would be a problem only in case someone had consciously set two stop bits. I think that's fine ...

Reply:
> that would be a problem only in case someone had consciously set two stop bits

My point, exactly.

> I think that's fine

Ok, then